src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
having to manually manage the precompile cache for correct behavior defeats the purpose of marking a package as opting in to precompilation though.